src/modules/0.0.22/aws_acm_list/index.ts

```diff
@@ -94,18 +103,24 @@ class CertificateMapper extends MapperBase<Certificate> {
       // Do not cloud create, just restore database
       await this.module.certificate.db.delete(es, ctx);
     },
-    read: async (ctx: Context, id?: string) => {
-      const client = (await ctx.getAwsClient()) as AWS;
-      if (id) {
-        const rawCert = await this.getCertificate(client.acmClient, id);
+    read: async (ctx: Context, arn?: string) => {
+      if (arn) {
+        const region = parseArn(arn).region;
+        const client = (await ctx.getAwsClient(region)) as AWS;
+
+        const rawCert = await this.getCertificate(client.acmClient, arn);
         if (!rawCert) return;
-        return this.certificateMapper(rawCert);
+        return this.certificateMapper(rawCert, region);
       } else {
-        const rawCerts = (await this.getCertificates(client.acmClient)) ?? [];
+        const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
         const out = [];
-        for (const rawCert of rawCerts) {
-          const cert = this.certificateMapper(rawCert);
-          if (cert) out.push(cert);
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const rawCerts = (await this.getCertificates(client.acmClient)) ?? [];
+          for (const rawCert of rawCerts) {
+            const cert = this.certificateMapper(rawCert, region);
+            if (cert) out.push(cert);
+          }
```

Comment:
let's do this in parallel with `await.Promise.all()` like here https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_api_gateway/index.ts#L79L91

Reply:
Cool way. I'll switch to that.